src/approaches/pg3_approach.py

```diff
@@ -46,14 +47,39 @@ def __init__(self, initial_predicates: Set[Predicate],
     def get_name(cls) -> str:
         return "pg3"
 
-    def _predict(self, state: State, atoms: Set[GroundAtom],
-                 goal: Set[GroundAtom]) -> _GroundNSRT:
-        del state  # unused
+    def predict(self, atoms: Set[GroundAtom],
```

Comment:
this can be a private method (add underscore). we can also make it more specific --  maybe `_predict_ground_nsrt`?